src/createStore.js

```diff
@@ -22,9 +22,11 @@ export var ActionTypes = {
  * @param {Function} reducer A function that returns the next state tree, given
  * the current state tree and the action to handle.
  *
- * @param {any} initialState The initial state. You may optionally specify it
+ * @param {Object} [initialState] The initial state. You may optionally specify it
```

Comment:
Nope. Only if you use `combineReducers`. Otherwise it can be anything, an array, an Immutable object, etc.